File: soa-spacemarine-java/src/main/java/ru/itmo/soa/soaspacemarinejava/repository/StarShipRepository.java
package ru.itmo.soa.soaspacemarinejava.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.itmo.soa.soaspacemarinejava.domain.SpaceMarine;
import ru.itmo.soa.soaspacemarinejava.domain.StarShip;

@Repository
public interface StarShipRepository extends JpaRepository<StarShip, Long> {

    StarShip findBySpaceMarine(SpaceMarine spaceMarine);
}

File: soa-spacemarine/src/main/kotlin/ru/itmo/soa/soaspacemarine/repository/StarShipRepository.kt
package ru.itmo.soa.soaspacemarine.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.itmo.soa.soaspacemarine.domain.SpaceMarine
import ru.itmo.soa.soaspacemarine.domain.StarShip

@Repository
interface StarShipRepository : JpaRepository<StarShip, Long> {

    fun findBySpaceMarine(spaceMarine: SpaceMarine): StarShip?
}